File: eAnimalShelter/eAnimalShelter.Services/Database/VolunteerAssignment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Services.Database
{
    public class VolunteerAssignment
    {
        [Key]
        public int AssignmentId { get; set; }
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }  = null!;
        public int ActivityId { get; set; }
        [ForeignKey("ActivityId")]
        public VolunteerActivity Activity { get; set; } = null!;
        public DateTime AppliedAt { get; set; }
        public string? ApplicationNote { get; set; }
        public AssignmentStatus Status { get; set; }
        public string? AdminResponseReason { get; set; }
        public decimal HoursWorked { get; set; }
    }


}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/volunteer_assignment.dart
class VolunteerAssignment {
  int? assignmentId;
  int? userId;
  String? userName;

  int? activityId;
  String? activityTitle;

  DateTime? appliedAt;

  String? applicationNote;
  int? status;

  double? hoursWorked;

  String? email;
  String? phoneNumber;
  String? adminResponseReason;

  VolunteerAssignment({
    this.assignmentId,
    this.userId,
    this.userName,
    this.activityId,
    this.activityTitle,
    this.appliedAt,
    this.applicationNote,
    this.status,
    this.hoursWorked,
    this.email,
    this.phoneNumber,
    this.adminResponseReason
  });

  factory VolunteerAssignment.fromJson(
    Map<String, dynamic> json,
  ) {
    return VolunteerAssignment(
      assignmentId: json["assignmentId"],
      userId: json["userId"],
      userName: json["userName"],
      activityId: json["activityId"],
      activityTitle: json["activityTitle"],
      appliedAt: json["appliedAt"] != null
          ? DateTime.parse(json["appliedAt"])
          : null,
      applicationNote: json["applicationNote"],
      status: json["status"],
      hoursWorked:
          (json["hoursWorked"] ?? 0).toDouble(),
      email: json["email"],
      phoneNumber: json["phoneNumber"],
      adminResponseReason: json["adminResponseReason"]
    );
  }
}